modules/fedimint-mint-client/src/lib.rs

```diff
@@ -650,23 +650,33 @@ impl ClientModule for MintClientModule {
         true
     }
 
-    async fn create_sufficient_input(
+    async fn create_final_inputs_and_outputs(
         &self,
         dbtx: &mut DatabaseTransaction<'_>,
         operation_id: OperationId,
-        min_amount: Amount,
-    ) -> anyhow::Result<Vec<ClientInput<MintInput, MintClientStateMachines>>> {
-        self.create_input(dbtx, operation_id, min_amount).await
-    }
+        input: Amount,
+        output: Amount,
+    ) -> anyhow::Result<(
+        Vec<ClientInput<MintInput, MintClientStateMachines>>,
+        Vec<ClientOutput<MintOutput, MintClientStateMachines>>,
+    )> {
+        let (inputs, selected_input) = self
```

Comment:
`selected_input` is a very bad name for the `selected_input_amount`, as mentioned before this should just be calculated here for readability.

Reply:
Done.